src/components/contract-functions/contract-function.tsx

```diff
@@ -228,9 +229,19 @@ export const ContractFunctionsPanel: React.FC<ContractFunctionsPanelProps> = ({
     return fnsOrEvents.filter((fn) => !("stateMutability" in fn)) as AbiEvent[];
   }, [fnsOrEvents]);
 
+  // Load state from the URL
+  const router = useRouter();
+  const _itemName = router.query.name;
+  const _item = fnsOrEvents.find((o) => o.name === _itemName);
   const [selectedFunction, setSelectedFunction] = useState<
     AbiFunction | AbiEvent
-  >(fnsOrEvents[0]);
+  >(_item ?? fnsOrEvents[0]);
+
+  // Set the active tab to Write or Read depends on the `_item`
+  const _defaultTabIndex =
+    _item && "stateMutability" in _item && _item["stateMutability"] === "view"
```

Comment:
I think this needs to be either `view` or `pure` (both show up in "read")

Reply:
thanks! lemme take a look